Lab1

VarRoman opened the pull request.

VarRoman committed:
Added Controller level file, exception and validator

Poluianov-Vladyslav committed:
Test and Service

Ekatereana commented on src/main/java/com/example/cosmocatsmarket/domain/Product.java:
```diff
@@ -0,0 +1,13 @@
+@Data
+@Builder
+public class Product {
+    Long id;
```
Long id -- погана ідея. писала в чаті чому. коротко -- так легко прогнозувати ваші id для зловмисників перебором

Ekatereana commented on src/main/java/com/example/cosmocatsmarket/dto/ProductDto.java:
```diff
@@ -0,0 +1,24 @@
+
+@Value
+@Builder(toBuilder = true)
+@Jacksonized
```
немає Validation Group

Ekatereana commented on src/main/java/com/example/cosmocatsmarket/dto/ProductEntry.java:
```diff
@@ -0,0 +1,15 @@
+@Builder
+@Jacksonized
+public class ProductEntry {
+    Long id;
```
аналогічно

VarRoman committed:
Solved the problem with Long id

VarRoman committed:
Finished extra task for a greater evaluation.
Added ProductRatingService, which accesses an external service to get a product rating

VarRoman commented:
Виправили помилки на які Ви вказали, а також, щоб отримати трішки більше балів за першу лабу та закрити предмет на прохідний бал зробили ще додаткове завдання(якщо це можливо, то зарахуйте, будь ласка, якщо ні, то ми зрозуміємо). Додали ProductRatingService, який звертається до зовнішнього сервісу для отримання рейтингу продуктів. Для моків відповідей сервісу було використано Wiremock. Для отримання продуктів з рейтингом потрібно додати параметр .../products?with-rating=true
